### PR TITLE
feat(payment): add 'other' payment channel in addition to email & phone

### PR DESCRIPTION
**Pull request by blaggacao**
# Context

- Phone payment channels are for MPesa (and MPesa only), they work by a wired way I don't fully understand and probably shouldn't be called 'Phone' for that reason.
- Email payment channels suppose to send the payment link via Email
- But what about sending the payment url via Whatsapp? Or integrating it into a chat bot?

# Proposed Solution

- Free the UI on it's restricive dependence on "not phone"
- Make it "email or default" instead
- Add 'Link' payment channel to represent that only the link generation is sought after but otherwise identical to the 'Email' channel but without [accidentially) sending emails.

### CODE REVIEW

**Comment by blaggacao** [comment]
potentially superseded by https://github.com/frappe/payments/issues/52 (only _"potentially"_ as of now)

File: erpnext/accounts/doctype/payment_gateway_account/payment_gateway_account.json [hunk 1]
```diff
@@ -69,12 +70,12 @@
    "fieldname": "payment_channel",
    "fieldtype": "Select",
    "label": "Payment Channel",
-   "options": "\nEmail\nPhone"
+   "options": "\nEmail\nPhone\nLink"
```
Comment (blaggacao) [inline erpnext/accounts/doctype/payment_gateway_account/payment_gateway_account.json]: Could also be named `Other`, wdyt?

**Comment by deepeshgarg007** [comment]
@blaggacao Can you fix linter check and update the PR?

**Comment by blaggacao** [comment]
@deepeshgarg007 ok, done, thanks for the review. Please proceed. Apologies for the delay. :handshake:

**Comment by blaggacao** [comment]
I unfortunately missed the stale-bot over the holidays. @deepeshgarg007, could you maybe reopen?

**Pull request closed without merging**